### PR TITLE
Generate only one full-courses.json

### PR DESCRIPTION
**Pull request by SamChou19815**
### Summary 

Right now we have two set of course json: one in the functions folder of filtered courses, one in assets folder as `full-courses.json`. However, we only need one: `full-courses.json`, since we are already only using course data of the latest offerings.

This PR unifies the script to generate only one and removes the others. I updated all the usages to use the single source of truth.

### Test Plan 

`npm run req-gen`. It only generates small amount of changes, which shows that the change is good. (Small number of changes are expected, since cornell updates the roster regularly and new courses might be added in the semester.)

### Notes

I also make the course json no longer collapse on a single line. It can make future changes and review easy.

### CODE REVIEW

File: src/requirements/courses-json-generator.ts [hunk 1]
```diff
@@ -1,17 +1,211 @@
...
-import { courseCodeToDataAndOfferedMap } from './filtered-course-data';
 import { Course } from './types';
 
-const fullCourseJson = (() => {
-  const json: Record<number, Course & { roster: string }[]> = {};
+const PREFIX = 'https://classes.cornell.edu/api/2.0';
+
+const wait = (time: number) => new Promise<void>(resolve => setTimeout(() => resolve(), time));
+
+const getSemesters = async (): Promise<readonly string[]> => {
```
Comment (tcho6319) [inline src/requirements/courses-json-generator.ts:13]: Can you document this code (like each `const` or at least in the form of a spec at the top of this script) to show that we're ultimately getting the latest offering of a course and then combining them into the full-courses.json?

**Review by tcho6319 — CHANGES_REQUESTED** [review]
Thanks Sam! This code clean-up definitely makes the course fetching process simpler for other devs to follow. I requested some documentation but once that's addressed, this PR should be g2g.

File: package.json [hunk 2]
```diff
@@ -5,7 +5,6 @@
   "scripts": {
     "serve": "vue-cli-service serve",
     "build": "vue-cli-service build --mode production",
-    "fetch-courses": "ts-node -T -P tsconfig.node.json src/requirements/fetcher.ts",
     "courses-gen": "ts-node -T -P tsconfig.node.json src/requirements/courses-json-generator.ts",
```
[the remark this answers is not in the document]
Comment (SamChou19815) [reply package.json:8]: Yes

**Review by tcho6319 — APPROVED** [review]
LGTM ty Sam!

**Pull request merged**